### PR TITLE
Add unreads property to rooms in user.info endpoint

### PR DESCRIPTION
**Pull request by MarcosSpessatto**
Related to the forum post: https://forums.rocket.chat/t/api-user-info-unread-messages-and-count-of/1152

### CODE REVIEW

**Comment by Maqsyo** [comment]
So there's no progress on this, mh? :(

**Comment by Kakoum** [comment]
Anyone have news?
Is there no way today to retrieve a user's list of unread messages?

:(

File: app/models/server/models/Subscriptions.js [hunk 1]
```diff
@@ -33,6 +33,51 @@ export class Subscriptions extends Base {
...
+					as: 'room',
+				},
+			},
+			{ $unwind: '$room' },
+			{
+				$lookup: {
+					from: 'rocketchat_message',
+					localField: 'rid',
+					foreignField: 'rid',
+					as: 'unreads',
```
Comment (rodrigok) [inline app/models/server/models/Subscriptions.js:54]: Use [pipeline](https://docs.mongodb.com/manual/reference/operator/aggregation/lookup/#lookup-join-pipeline) here to filter and get only the unread messages

Comment (rodrigok) [reply app/models/server/models/Subscriptions.js:54]: Use pipeline on room lookup as well to get only the `lm` field

File: app/api/server/v1/users.js [hunk 2]
```diff
@@ -193,22 +193,24 @@ API.v1.addRoute('users.info', { authRequired: true }, {
 		if (!result || result.count() !== 1) {
 			return API.v1.failure('User not found.');
 		}
+
 		const [user] = result.fetch();
 		const myself = user._id === this.userId;
 		if (fields.userRooms === 1 && (myself || hasPermission(this.userId, 'view-other-user-channels'))) {
-			user.rooms = Subscriptions.findByUserId(user._id, {
+			user.rooms = Promise.await(Subscriptions.findByUserIdWithUnreadMessagesCount(user._id, {
```
Comment (rodrigok) [inline app/api/server/v1/users.js:200]: We should have a param to decide whenever check for the unreads as well.

**Comment by Boby** [comment]
jumping on this, is there any api on this?

**Comment by WithAlex** [comment]
any news ?

**Comment by alba-simedia** [comment]
If api is not available, it would be a solution use notifications?

**Comment by tassoevan** [comment]
Replaced by #20905.

**Pull request closed without merging**